src/kms_sink.cpp

```diff
@@ -31,17 +33,21 @@ KMSSink::KMSSink(const std::string& connectorName) {
    * pick the first connected connector or, if no connector is connected,
    * the first connector with unknown status.
    */
+  bool breakLoop = false;
   for (const DRM::Connector& conn : dev_.connectors()) {
```

Comment:
I see this issue now in Sonar, if you make this it's own function, the whole for loop (call it something like findRequestedConnector) and use returns instead of breaks you fix this... This is a MISRA one too which is related to AUTOSAR standards so it would be nice to fix...

Reply:
Behaviour has changed as is here because if the first `breakLoop = true` is hit, we still execute the next if statement, which is not what we want...

Reply:
Most complaints about goto and break usage can be refactored by using smaller functions and returns. Things like returns, break and goto are similar.